src/core/dbus-execute.c

```diff
@@ -3578,6 +3609,41 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (STR_IN_SET(name, "StateDirectoryQuota", "CacheDirectoryQuota", "LogsDirectoryQuota")) {
+                uint64_t quota_absolute = UINT64_MAX;
+                uint32_t quota_scale = UINT32_MAX;
+                int quota_enforce;
+
+                r = sd_bus_message_read(message, "(tub)", &quota_absolute, &quota_scale, &quota_enforce);
+                if (r < 0)
+                        return r;
+
+                ExecDirectoryType dt;
+                if (streq(name, "StateDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_STATE;
+                else if (streq(name, "CacheDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_CACHE;
+                else if (streq(name, "LogsDirectoryQuota"))
+                        dt = EXEC_DIRECTORY_LOGS;
+                else
+                        assert_not_reached();
+
+                if (!quota_enforce) {
```

Comment:
i'd invert the two if branches, to put the "positive" branch first, and the negative branch seconds, seems more obvious to me. (and would save one characte... ;-))